collated/main/A0139772U.md

```diff
@@ -0,0 +1,2562 @@
+
+	public static final String COMMAND_WORD = "add";
+
+	public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a task to WhatNow. \n"
+			+ "Parameters: \"TASK_NAME\" [t/TAG]...\n"
+	        + "Parameters: \"TASK_NAME\" [on/by/from] [today/tomorrow/DATE] [to] [today/tomorrow/DATE] [t/TAG]...\n"
+			+ "Parameters: \"TASK_NAME\" [by/at/from] [TIME] [till/to] [TIME] [t/TAG]...\n"
+			+ "Example: \n"
+			+ COMMAND_WORD + " \"Buy groceries\" on 23/2/2017 t/highPriority\n"
+			+ COMMAND_WORD + " \"Buy dinner\" at 6pm t/highPriority\n"
+			+ COMMAND_WORD + " \"Lesson\" on 24/2/2017 from 8.30am to 4:30pm t/lowPriority\n"
+			+ COMMAND_WORD + " \"Submit homework\" by tomorrow 12pm t/lowPriority\n";
+
+	public static final String MESSAGE_SUCCESS = "New task added: %1$s";
+	public static final String MESSAGE_DUPLICATE_TASK = "This task already exists in WhatNow";
+	private static final String STATUS_INCOMPLETE = "incomplete";
+
+	private final Task toAdd;
+	
+```
+###### \java\seedu\whatnow\logic\commands\DeleteCommand.java
+``` java
+package seedu.whatnow.logic.commands;
+
+import seedu.whatnow.commons.core.Messages;
+import seedu.whatnow.commons.core.UnmodifiableObservableList;
+import seedu.whatnow.model.task.ReadOnlyTask;
+import seedu.whatnow.model.task.Task;
+import seedu.whatnow.model.task.UniqueTaskList.DuplicateTaskException;
+import seedu.whatnow.model.task.UniqueTaskList.TaskNotFoundException;
+
+/**
+ * Deletes a task identified using it's last displayed index from WhatNow.
+ */
+public class DeleteCommand extends UndoAndRedo {
+
+	public static final String COMMAND_WORD = "delete";
+
+	public static final String MESSAGE_USAGE = COMMAND_WORD
+			+ ": Deletes the task identified by the index number used in the last task listing.\n"
+			+ "Parameters: INDEX (must be a positive integer)\n"
+			+ "Example: " + COMMAND_WORD + " 1";
+
+	public static final String MESSAGE_DELETE_TASK_SUCCESS = "Deleted Task: %1$s";
+
+	private static final String TASK_TYPE_FLOATING = "todo";
+
+	private final int targetIndex;
+	private final String taskType;
+
+	public DeleteCommand(String taskType, int targetIndex) {
+		this.targetIndex = targetIndex;
+		this.taskType = taskType;
+	}
+
+
+	@Override
+	public CommandResult execute() {
+
+		UnmodifiableObservableList<ReadOnlyTask> lastShownList;
+
+		if (taskType.equals(TASK_TYPE_FLOATING)) {
+			lastShownList = model.getCurrentFilteredTaskList();
+		} else {
+			lastShownList = model.getCurrentFilteredScheduleList();
+		}
+		if (lastShownList.size() < targetIndex) {
+			indicateAttemptToExecuteIncorrectCommand();
+			return new CommandResult(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+		}
+		ReadOnlyTask taskToDelete = lastShownList.get(targetIndex - 1);
+
+		assert model != null;	 
+		try {
+			model.deleteTask(taskToDelete);
+			model.getUndoStack().push(this);
+			model.getDeletedStackOfTasks().push(taskToDelete);
+		} catch (TaskNotFoundException pnfe) {
```

Comment:
`pnfe` stood for `PersonNotFoundException` in addressbook. you should rename this variable appropriately since the context is no longer persons or addresses